internal/handlers/common/projection.go

```diff
@@ -234,3 +258,161 @@ func filterFieldArrayElemMatch(k1 string, doc, conditions *types.Document, docVa
 	}
 	return
 }
+
+// filterFieldArraySlice implements $slice projection query.
+func filterFieldArraySlice(docValue *types.Array, projectionValue any) (*types.Array, error) {
```

Comment:
Overall, looks overcomplicated. Can we somehow simplify that function by splitting it or using earlier returns? For example, two cases of the `switch projectionValue.(type)` below look almost unrelated, so probably could be extracted into their own functions